src/tools/bzlmod/utils.bzl

```diff
@@ -122,13 +169,17 @@ def parse_bazel_module_repos(ctx, lockfile_path):
 
     lockfile = json.decode(ctx.read(lockfile_path))
     repos = []
-    for _, module in lockfile["moduleDepGraph"].items():
-        if "repoSpec" in module and module["repoSpec"]["ruleClassName"] == "http_archive":
-            repo_spec = module["repoSpec"]
-            attributes = repo_spec["attributes"]
-            repo_name = _module_repo_name(module)
-            repos.append(repo_name)
-    return repos
+    for url in lockfile["registryFileHashes"].keys():
```

Comment:
could we also introduce some backwards-compatible logic here (similar to line 56+) and avoid checking in the dist lockfile? I'm worried that having to maintain the dist lockfile in addition could make cherry-picking even harder.

Reply:
I implemented module resolution in Starlark, which turned out to be not so bad with `buildozer`. This will fail without https://github.com/bazelbuild/bazel/issues/22287 though.

Reply:
oh wow, I hadn't realized that was what this required. But kudos for cranking it out so quickly :)